File: frontend/src/api/treatments.ts
import { apiClient } from '@/api/client'
import type { Treatment, PaginatedResponse } from '@/types'

export interface TreatmentListParams {
  page?: number
  size?: number
  status?: 'active' | 'completed'
}

export interface TreatmentPayload {
  name: string
  date_start: string
  days: number
  receipt: string
  body_region?: string | null
}

export async function listTreatments(params?: TreatmentListParams): Promise<PaginatedResponse<Treatment>> {
  const response = await apiClient.get('/api/v1/treatments/', { params })
  return response.data
}

export async function getTreatment(id: number): Promise<Treatment> {
  const response = await apiClient.get(`/api/v1/treatments/${id}`)
  return response.data
}

export async function createTreatment(data: TreatmentPayload): Promise<Treatment> {
  const response = await apiClient.post('/api/v1/treatments/', data)
  return response.data
}

export async function updateTreatment(id: number, data: TreatmentPayload): Promise<Treatment> {
  const response = await apiClient.put(`/api/v1/treatments/${id}`, data)
  return response.data
}

export async function deleteTreatment(id: number): Promise<void> {
  await apiClient.delete(`/api/v1/treatments/${id}`)
}

File: backend/app/api/treatments.py
from typing import Optional

from fastapi import APIRouter, Depends, Query

from app.api.dependencies import get_current_user, get_treatment_service
from app.application.commands import CreateTreatmentCommand, UpdateTreatmentCommand
from app.application.pagination import calculate_pages
from app.application.treatment_service import TreatmentAppService
from app.domain.entities import User
from app.schemas.treatment import TreatmentCreate, TreatmentListResponse, TreatmentResponse, TreatmentUpdate

router = APIRouter()


@router.get("/", response_model=TreatmentListResponse)
async def list_treatments(
    page: int = Query(1, ge=1),
    size: int = Query(20, ge=1, le=100),
    status_filter: Optional[str] = Query(None, alias="status"),
    sort: str = Query("-date_start"),
    current_user: User = Depends(get_current_user),
    service: TreatmentAppService = Depends(get_treatment_service),
):
    items, total = await service.list(current_user.id, status_filter, sort, page, size)
    pages = calculate_pages(total, size)
    return TreatmentListResponse(items=[TreatmentResponse.model_validate(t) for t in items], total=total, page=page, size=size, pages=pages)


@router.get("/{treatment_id}", response_model=TreatmentResponse)
async def get_treatment(
    treatment_id: int,
    current_user: User = Depends(get_current_user),
    service: TreatmentAppService = Depends(get_treatment_service),
):
    return TreatmentResponse.model_validate(await service.get(treatment_id, current_user.id))


@router.post("/", response_model=TreatmentResponse, status_code=201)
async def create_treatment(
    data: TreatmentCreate,
    current_user: User = Depends(get_current_user),
    service: TreatmentAppService = Depends(get_treatment_service),
):
    cmd = CreateTreatmentCommand(
        date_start=data.date_start, name=data.name, days=data.days,
        receipt=data.receipt, body_region=data.body_region,
    )
    return TreatmentResponse.model_validate(await service.create(current_user.id, cmd))


@router.put("/{treatment_id}", response_model=TreatmentResponse)
async def update_treatment(
    treatment_id: int,
    data: TreatmentUpdate,
    current_user: User = Depends(get_current_user),
    service: TreatmentAppService = Depends(get_treatment_service),
):
    cmd = UpdateTreatmentCommand(**data.model_dump(exclude_unset=True))
    return TreatmentResponse.model_validate(await service.update(treatment_id, current_user.id, cmd))


@router.delete("/{treatment_id}", status_code=204)
async def delete_treatment(
    treatment_id: int,
    current_user: User = Depends(get_current_user),
    service: TreatmentAppService = Depends(get_treatment_service),
):
    await service.delete(treatment_id, current_user.id)
